src/test/java/io/cdap/plugin/sfmc/sink/RecordDataExtensionRowConverterTest.java

```diff
@@ -0,0 +1,111 @@
+    column.setName("name");
+    column.setModified("date", false);
+    column.setPrecision(0);
+    column.setScale(100);
+    columnList.add(column);
+    Map<String, ETDataExtensionColumn> columns = new HashMap<>();
+    columns.put("tableName", column);
+    String externalKey = "externalKey";
+    DataExtensionInfo dataExtensionInfo = new DataExtensionInfo(externalKey, columnList);
+    RecordDataExtensionRowConverter recorddataextensionrowconverter =
+      new RecordDataExtensionRowConverter(dataExtensionInfo, true);
+    Assert.assertEquals(recorddataextensionrowconverter.getType(type, fieldName, val1), strVal);
+    Assert.assertEquals(recorddataextensionrowconverter.getType(type2, fieldName2, val2), String.valueOf(val2));
+    Assert.assertEquals(recorddataextensionrowconverter.getType(type3, fieldName3, val3), ((Enum) val3).name());
+  }
+
+  @Test
+  public void testTransform() throws ETSdkException {
+    List<ETDataExtensionColumn> columnList = new ArrayList<>();
+    ETDataExtensionColumn column1 = new ETDataExtensionColumn();
+    column1.setName("storeid");
+    column1.setName("emailid");
+    column1.setName("timestamp");
+    column1.setName("price");
+    ETDataExtensionColumn column2 = new ETDataExtensionColumn();
+    column2.setName("date");
+    columnList.add(column1);
+    columnList.add(column2);
+    Schema inputSchema = Schema.recordOf("record",
+                                         Schema.Field.of("storeid", Schema.of(Schema.Type.STRING)),
+                                         Schema.Field.of("emailid", Schema.of(Schema.Type.STRING)),
+                                         Schema.Field.of("price", Schema.nullableOf(Schema.decimalOf(4, 2))),
+                                         Schema.Field.of("timestamp",
+                                                         Schema.nullableOf(Schema.of
+                                                           (Schema.LogicalType.TIMESTAMP_MICROS))),
+                                         Schema.Field.of("date", Schema.of(Schema.LogicalType.DATE)));
+    DataExtensionInfo dataExtensionInfo = new DataExtensionInfo("DE", columnList);
+    RecordDataExtensionRowConverter recordDataExtensionRowConverter =
+      new RecordDataExtensionRowConverter(dataExtensionInfo, false);
+    StructuredRecord structuredRecord = Mockito.mock(StructuredRecord.class);
+    Mockito.when(structuredRecord.getSchema()).thenReturn(inputSchema);
+    Mockito.when(structuredRecord.get("emailid")).thenReturn("abc@123");
+    LocalDate date = LocalDate.of(2011, 01, 01);
+    Mockito.when(structuredRecord.get("date")).thenReturn(date);
+    Mockito.when(structuredRecord.getDate("date")).thenReturn(date);
+    Mockito.when(structuredRecord.get("price")).thenReturn(2000);
+    Mockito.when(structuredRecord.getDecimal("price")).thenReturn(BigDecimal.valueOf(2000));
+    Assert.assertFalse(recordDataExtensionRowConverter.transform(structuredRecord).getColumn("price").isEmpty());
```

Comment:
check the actual value, not just non-empty

Reply:
Done

Reply:
remove redundant Assert.assertFalse()

Reply:
Removed